docs/azrepos-users-and-tokens.md

```diff
@@ -0,0 +1,226 @@
+# Azure Repos: Access Tokens and Accounts
+
+## Different Credential Types
+
+The Azure Repos host provider supports creating multiple types of credential:
+
+- Personal Access Token
+- Azure Access Token
+
+### Personal Access Tokens
+
+Historically, the only option supported by the Azure Repos host provider was
+Azure DevOps Personal Access Tokens (PATs).
+
+These PATs are only used by Azure DevOps, and must be [managed through the Azure
+DevOps user settings page](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page).
+
+PATs have a limited lifetime and new tokens must be created once they expire. In
+Git Credential Manager, when a PAT expired (or was manually revoked) this
+resulted in a new authentication prompt.
+
+### Azure Access Tokens
+
+Azure Access Tokens is the generic term for tokens issued by Azure Active
+Directory for either Work and School Accounts (AAD tokens) or Personal Accounts
+(Microsoft Account/MSA tokens).
+
+Azure DevOps supports Git authentication using Azure Access Tokens as well as
+PATs. Azure Access Tokens created by Git Credential Manager are scoped to Azure
+DevOps only.
+
+Unlike PATs, Azure Access Tokens get automatically refreshed and renewed as long
+as you are actively using them to perform Git operations.
+
+These tokens are also securely shared with other Microsoft developer tools
+including the Visual Studio IDE and Azure CLI. This means that as long as you're
+using Git or one of these tools with the same account, you'll never need to
+re-authenticate due to expired tokens!
+
+## User Accounts
+
+In versions of Git Credential Manager that support Azure Access Tokens, the user
+account used to authenticate for a particular Azure DevOps organization will now
+be remembered.
+
+The first time you clone, fetch or push from/to an Azure DevOps organization you
+will be prompted to sign-in and select a user account. Git Credential Manager
```

Comment:
I know on the Azure DevOps side, they say "identity" for the thing that represents a user. Genuine question, is that the same thing as "account" here (and if so, should we use their terminology)?

Reply:
We can say "identity" if we want to be consistent with Azure DevOps.

In AAD there are users and accounts (an account != user.. one can span tenants and clouds, the other is in a particular cloud+tenant).

In Git there is only a "username".

Reply:
I guess technically, you're selecting an Azure _account_ when you sign-in, and not an Azure DevOps _identity_.. Just happens that Azure accounts are mapped to an AzDO identity.

Aside: there's also a _FUN_ thing that if you've never signed-in to the web front-end of a particular Azure DevOps organisation, you cannot call any AzDevOps APIs because the "identity" doesn't exist yet - it's only materialised on first sign-in on the web. 🙄

Reply:
> if you've never signed-in to the web front-end of a particular Azure DevOps organisation, you cannot call any AzDevOps APIs

Let me know if you are seeing this - it means that someone forgot to turn on a feature flag.

Reply:
I know this was the case back when GCM Core was first out. I remember hitting this exact issue and having to include support for surfacing human readable error messages (telling you do go log in via the web first) from Azure DevOps' REST API endpoints.